### ISSUE TITLE
Standardize `(define ...)` naming

### ISSUE THREAD

**Issue by kantai** [blockstack/blockstack-core#968]
We currently have several define functions:

* `(define ...)` -- defines a private function or variable
* `(define-public ...)` -- defines a public function
* `(define-map ...)` -- defines a map
* `(define-read-only ...)` -- defines a public read only function

`(define ...)` seems out of place --- should we rename to `(define-private ...)` ?

**Comment by jcnelson** [comment]
Do we want to consider making them a bit more terse?  Something like:

* `(defun ...)` -- your garden-variety function declaration
* `(defpub ...)` -- a publicly-callable function
* `(defmap ...)` -- define a map
* `(defro .. )` -- define a read-only publicly-callable function

I'm not picky just as long as the terms bring clarity ;)

**Comment by zone117x** [comment]
I'm in favor of something more terse, as long as it's still clear what the abbreviations or acronyms are. 
`defpub` and `defmap` seem clear, while `defro` and `defun` not as much.

**Comment by kantai** [comment]
Yep -- something more terse is fine with me --

Maybe:

`def-pub`
`def-map`
`def-priv`
`def-read-only`

Read only is still kind of verbose, but I think any abbreviation there would be pretty unclear

**Comment by jcnelson** [comment]
`rdonly?`

**Comment by kantai** [comment]
```
(define) -> (define-private)
(define-public) -> ...
(define-read-only) -> ..
```

**Comment by moxiegirl** [comment]
Do you really want the dashes in there? 

`defprivf`
`defpubf`
`defrof`

You can also consider putting the unique portion as the suffix for ease in scanning.

`privatedef`
`publicdef`
`readonlydef`

**Comment by kantai** [comment]
Closed by #1084
